calendar@deeppradhan/files/calendar@deeppradhan/desklet.js

```diff
@@ -0,0 +1,252 @@
+			this.date = Calendar.dateMonthAdd(this.date, -1);
+			this.updateCalendar();
+		}));
+		this.buttonNext.connect("clicked", Lang.bind(this, function() {
+			this.date = Calendar.dateMonthAdd(this.date, 1);
+			this.updateCalendar();
+		}));
+
+		this.tooltipMonth = new Tooltips.Tooltip(this.buttonMonth);
+		this.tooltipPrevious = new Tooltips.Tooltip(this.buttonPrevious,
+				_("Previous month..."));
+		this.tooltipNext = new Tooltips.Tooltip(this.buttonNext,
+				_("Next month..."));
+
+		this.tableMonth.add(this.buttonPrevious, {row: 0, col: 0});
+		this.tableMonth.add(this.buttonMonth, {row: 0, col: 1, colSpan: 5});
+		this.tableMonth.add(this.buttonNext, {row: 0, col: 6});
+
+		// Create buttons with labels (with tooltips) for days
+		for (let i = 0; i < 31; i++) {
+			this.labelDays[i] = new St.Label();
+			this.labelDays[i].style = STYLE_LABEL_DAY;
+			this.labelDays[i].set_text(String(i + 1));
+		}
+
+		//////// Calendar Layout ////////
+		// Set Desklet header
+		this.setHeader("Calendar");
+
+		this.updateCalendar();
+	},
+
+	// Called on user clicking the desklet
+	on_desklet_clicked: function(event) {
+		this.date = new Date();
+		this.updateCalendar();
+	},
+	
+	on_desklet_removed: function() {
+		Mainloop.source_remove(this.timeout);
+	},
+	
+	// Refresh on change of settings
+	onSettingChanged: function() {
+		Mainloop.source_remove(this.timeout);
+		this.updateCalendar();
+	},
+
+	/* Method to update the Desklet layout*/
+	updateCalendar: function() {
+
+		let now = new Date();
+
+		this.lastUpdate = { fullYear: now.getFullYear(), month: now.getMonth(), date: now.getDate()};
+
+		//////// Today Panel ////////
+		this.labelDate.style = (now.getDay() === 0 ? "color: " + this.colourSundays + "; " : "")
+				+ (now.getDay() === 6 ? "color: " + this.colourSaturdays + "; " : "")
+				+ "font-size: 4em; " + STYLE_TEXT_CENTER;
+
+		if (now.getDay() === 0 || now.getDay() === 6)
+			this.labelDay.style = "color: " + (now.getDay() === 0 ?
+					this.colourSundays : this.colourSaturdays) + "; " + STYLE_TEXT_CENTER;
+
+		this.boxLayoutToday.remove_all_children();
+		if (this.showWeekday !== "off")
+			this.boxLayoutToday.add(this.labelDay);
+		this.boxLayoutToday.add(this.labelDate);
+		this.boxLayoutToday.add(this.labelMonthYear);
+		if (this.showTime)
+			this.boxLayoutToday.add(this.labelTime);
+
+		//////// Month Panel ////////
+		this.labelMonth.set_text(Calendar.MONTH_NAMES[this.date.getMonth()].substring(0, 3) + " " + this.date.getFullYear());
+
+		// Set weekday style
+		for (let i = 0; i < 7; i++)
+			this.labelWeekdays[i].style = STYLE_LABEL_DAY + (this.date.getFullYear() == now.getFullYear()
+					&& this.date.getMonth() == now.getMonth() && i == now.getDay() ?
+					" font-weight: bold;" : "") + (i === 0 ? " color: " + this.colourSundays + ";" : "")
+					+ (i === 6 ? " color: " + this.colourSaturdays + ";" : "");
+
+	// Remove all days
+		for (let i = 0; i < 31; i++)
+			this.tableMonth.remove_child(this.labelDays[i]);
```

Comment:
Sorry, should have seen this earlier, in ~/.xsession-errors I'm seeing some Clutter errors caused by this line.

```
(cinnamon:2990): Clutter-CRITICAL **: 20:03:50.386: clutter_actor_remove_child: assertion 'child->priv->parent != NULL' failed
```

I was able to fix it by changing the for loop to check for the parent first, if its `null`, `continue`.

```js
for (let i = 0; i < 31; i++) {
  let parent = this.labelDays[i].get_parent();
	if (!parent) {
		continue;
	}
	parent.remove_child(this.labelDays[i]);
}
```

I'm guessing some some of the `labelDays` actors don't have a parent when they're passed to `remove_child`.